qt/applications/workbench/workbench/plotting/figuremanager.py

```diff
@@ -283,6 +285,30 @@ def destroy(self, *args):
             # Gcf can get destroyed before the Gcf.destroy
             # line is run, leading to a useless AttributeError.
 
+    def on_home_clicked(self):
+        """
+        The matplotlib home button centres and scales the figure.
+        When we have visible errorbars, we should scale off of the bounds
+        of the error bars.
+        """
+        ax = self.canvas.figure.axes[0]
+        creation_args = ax.creation_args
+
+        min_bounds = []
+        max_bounds = []
+        for cargs in creation_args:
+            # Creation args do not necessarily contain POST_CREATION_ARGS
+            if (MantidAxKwargs.POST_CREATION_ARGS in cargs and
+                    cargs[MantidAxKwargs.POST_CREATION_ARGS][MantidAxKwargs.ERRORS_VISIBLE]):
+                for container in ax.containers:
+                    caps = container[1]
+                    if caps != ():
+                        min_bounds.extend(caps[0].get_ydata())
+                        max_bounds.extend(caps[1].get_ydata())
+        if min_bounds and max_bounds:
+            # Set ylim if we have caps
+            ax.set_ylim((min(min_bounds), max(max_bounds)))
```

Comment:
@TTitcombe I think MPL sets some margins on the limits, as there's always some space between the data and the borders of the plot. Currently, after showing the errors and going to the "home" view the data seems to end exactly at the border of the plot. 

Do you think it'd be possible to the margins back while errors are shown? I found [this](https://stackoverflow.com/a/42804403/2823526) which mentions this method `ax.set_ymargin(0.1)`, alternatively also gives this `ymax = data.max()+0.4*(data.max()-data.min())` for a 40% margin (I think about 10%-15% should be enough). Another possible route is to check how MPL does it, but that would involve some digging in the source

Reply:
Good idea. I've calculated the margin manually by adding padding by 5% of the range of the data to the top and bottom (`set_ymargin` didn't work)